docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+
+The ability to modify a block state coupled with the fact that a block can live within any other block (think of a paragraph within a group), creates a vast amount of potential states and style possibilities.
+
+#### From UI controls to HTML markup
+
+If you follow the [block tutorial](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/) you can learn up about the different parts of the [block API](https://developer.wordpress.org/block-editor/reference-guides/block-api/) presented here in more detail and also build your own block. This is an introduction to the general concepts of how a block can let users edit its state.
+
+To build a experience like the one described above a block author needs a few pieces:
+
+1. **An UI control**. It presents the user some choices, for example, be able to change the font size of the block. The control takes care of reading the data from the block (does this block has already a font size assigned?) and other data it needs (what are the font sizes an user can use in this block?). See available [component library](https://developer.wordpress.org/block-editor/reference-guides/components/).
+2. **A block attribute**. The block needs to hold data to know which modifications were applied to it: whether it has been given a font size already, for example. See how blocks can define [attributes](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/).
+3. **Access to style data**. A control may need external information about the styles available for a given block: the list of colors, or the list of font sizes, for example. These are called "style presets", as they are a preselection of styles usually defined by the theme, although WordPress provides some defaults. Check the [list of data](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/#settings) a theme can provide to the editor and how a block author can get access to it via [useSetting](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/#usesetting).
+4. **Serialize the user style into HTML markup**. Upon an user action, the block HTML markup needs to be updated accordingly (apply the proper class or inline style). This process is called serialization and it is the [edit, save](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/), and [render_callback](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/creating-dynamic-blocks/) responsibility: these functions take block data and convert it into HTML.
+
+In essence, these are the essential mechanics a block author needs to care about for their block to be able to be styled by the user. While this can be done completely manually, there's an API that automates this process for common style needs: Block Supports.
+
+#### Block Supports API
```

Comment:
Sharing feedback I've got privately:

1. Some folks shared interest in going deep down into _how this works_: the filters in use, pointing directly to the code where this happens, etc.
2. Point to the existence of some functions useful "skip serialization" mode (`__experimentalUseColorProps`, `__experimentalUseBorderProps`, `__experimentalGetSpacingClassesAndStyles`, etc).

I'll incorporate something for the second point directly in this PR.

As for the 1st point. My first instinct was to keep this doc as approachable as possible to block authors, so they don't feel that they need to know how the internals of this API work to implement their blocks. That shouldn't be a requirement. I also understand how that would be useful for core contributors. I'd like to hear thoughts on this, especially from other folks new to this area or that have interacted with this lately.

Reply:
I felt that the "styles" document might not be the right place to expand on how block supports work. Would it be better if in this document block supports are treated just like any "local/serializable" style and just link to their document for block authors that want to know more about them?

Reply:
I read the content here for block supports and it does make a lot of sense and have value but maybe it's better moved to the block supports document itself.

Reply:
> Would it be better if in this document block supports are treated just like any "local/serializable" style and just link to their document for block authors that want to know more about them?

This sounds like a good idea. Block supports are still in development, so the docs might need a roomy home. 

It's probably still relevant to this doc to emphasise that many block supports add inline styles to block containers, and that they're rendered to the page with the block.

Reply:
I won't add any further to "how block supports" work and the utility functions in use. This can be done in a different PR and probably can go in the block supports docs.